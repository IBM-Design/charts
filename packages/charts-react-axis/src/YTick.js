import React, { PropTypes } from 'react';

const YTick = (props) => {
  const { label, offset, width } = props;
  const transform = `translate(0,${offset})`;

  return (
    <g className="tick" transform={transform}>
      <path
        d={`M0,0 L${width},0`}
        fill="none"
        stroke="#CDDBF1"
        strokeWidth="1"
      />
      <text fill="#000" x="-9" y="0.5" dy="0.32em">
        {label}
      </text>
    </g>
  );
};

YTick.propTypes = {
  label: PropTypes.string.isRequired,
  offset: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default YTick;
