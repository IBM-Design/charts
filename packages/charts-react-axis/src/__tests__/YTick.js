import React from 'react';
import renderer from 'react-test-renderer';
import YTick from '../YTick';

describe('YTick Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <YTick
        label="Label"
        offset={250}
        width={960}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
