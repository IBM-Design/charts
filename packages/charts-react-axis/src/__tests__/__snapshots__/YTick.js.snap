exports[`YTick Component should render 1`] = `
<g
  className="tick"
  transform="translate(0,250)">
  <path
    d="M0,0 Lundefined,0"
    fill="none"
    stroke="#CDDBF1"
    strokeWidth="1" />
  <text
    dy="0.32em"
    fill="#000"
    x="-9"
    y="0.5">
    Label
  </text>
</g>
`;
